lib/src/DateRangePicker/DateRangePickerInput.tsx

```diff
@@ -0,0 +1,161 @@
+import { MaterialUiPickersDate } from '../typings/date';
+import { DateInputProps } from '../_shared/PureDateInput';
+import { CurrentlySelectingRangeEndProps } from './RangeTypes';
+import { mergeRefs, createDelegatedEventHandler, doNothing } from '../_helpers/utils';
+
+export const useStyles = makeStyles(
+  theme => ({
+    rangeInputsContainer: {
+      display: 'flex',
+      alignItems: 'center',
+      [theme.breakpoints.down('xs')]: {
+        flexDirection: 'column',
+      },
+    },
+    toLabelDelimiter: {
+      margin: '8px 0',
+      [theme.breakpoints.up('sm')]: {
+        margin: '0 16px',
+      },
+    },
+  }),
+  { name: 'MuiPickersDateRangePickerInput' }
+);
+
+export interface ExportedDateRangePickerInputProps {
+  toText?: React.ReactNode;
+}
+
+export interface DateRangeInputProps
+  extends ExportedDateRangePickerInputProps,
+    CurrentlySelectingRangeEndProps,
+    Omit<DateInputProps<RangeInput, DateRange>, 'forwardedRef'> {
+  startText: React.ReactNode;
+  endText: React.ReactNode;
+  forwardedRef?: React.Ref<HTMLDivElement>;
+  containerRef?: React.Ref<HTMLDivElement>;
+}
+
+export const DateRangePickerInput: React.FC<DateRangeInputProps> = ({
+  toText = 'to',
+  rawValue,
+  onChange,
+  onClick,
+  parsedDateValue: [start, end],
+  id,
+  open,
+  className,
+  containerRef,
+  forwardedRef,
+  currentlySelectingRangeEnd,
+  setCurrentlySelectingRangeEnd,
+  openPicker,
+  onFocus,
+  readOnly,
+  disableOpenPicker,
+  startText,
+  endText,
+  ...other
+}) => {
+  const utils = useUtils();
+  const classes = useStyles();
+  const startRef = React.useRef<HTMLInputElement>(null);
+  const endRef = React.useRef<HTMLInputElement>(null);
+
+  React.useEffect(() => {
+    if (!open) {
+      return;
+    }
+
+    if (currentlySelectingRangeEnd === 'start') {
+      startRef.current?.focus();
+    } else if (currentlySelectingRangeEnd === 'end') {
+      endRef.current?.focus();
+    }
+  }, [currentlySelectingRangeEnd, open]);
+
+  const handleStartChange = (date: MaterialUiPickersDate, inputString?: string) => {
+    if (date === null || utils.isValid(date)) {
+      onChange([date, end], inputString);
+    }
+  };
+
+  const handleEndChange = (date: MaterialUiPickersDate, inputString?: string) => {
+    if (date === null || utils.isValid(date)) {
+      onChange([start, date], inputString);
+    }
+  };
+
+  const openRangeStartSelection = () => {
+    if (setCurrentlySelectingRangeEnd) {
+      setCurrentlySelectingRangeEnd('start');
+    }
+    if (!disableOpenPicker) {
+      openPicker();
+    }
+  };
+
+  const openRangeEndSelection = () => {
+    if (setCurrentlySelectingRangeEnd) {
+      setCurrentlySelectingRangeEnd('end');
+    }
+    if (!disableOpenPicker) {
+      openPicker();
+    }
+  };
+
+  return (
+    <div
+      id={id}
+      className={clsx(classes.rangeInputsContainer, className)}
+      ref={mergeRefs([containerRef, forwardedRef])}
+    >
+      <KeyboardDateInput
```

Comment:
[Suggestion]: You might want to provide an option to disabled auto fill from browser.

Reply:
Do you have more details on the potential problem and solution? Thanks.

Reply:
So in the root input field component, just need to disable autoComplete :P
```
 <InputField autoComplete="off" ... />
```
:)

Reply:
You can pass a prop `inputProps={{ autoComplete: "off" }}` right to the component. But this API will be changed to `renderInput={props => <TextField {...props} />}`

Reply:
What should be the default behavior? Displaying the browser's suggestion on top of the popup could be an issue but we won't display the popup by default as in https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/datepicker-dialog.html. I was proposing two optional props `openOnFocus` and `openOnClick` to have the popup open by default. In such case, should we change the default autocomplete? I'm not sure. We do change is with the Autocomplete, and had no people complaining, but still?

I'm in favor of not changing the current tradeoff.

Reply:
The reason behind suggesting it being off is one scenario we encountered before:
When we have both suggested auto fill & calendar deployed(opened), it was not a very good user experience to keep both.
As mentioned, it is an optional change. I'm happy to see how the current API goes and maybe the autoOn is a more favoring solution :P